src/object/background.cpp

```diff
@@ -396,94 +505,32 @@ Background::draw(DrawingContext& context)
-  {"forest_para3_bottom.png", "forest/forest_para3_bottom.png"},
-  {"nighthills.png", "forest/nighthills.png"},
-  {"ghostforest.jpg", "ghostforest/ghostforest.jpg"},
-  {"ghostforest_grave.png", "ghostforest/ghostforest_grave.png"},
-  {"ghostforest_para.png", "ghostforest/ghostforest_para.png"},
-  {"cave2.jpg", "ice_cave/cave2.jpg"},
-  {"darkcave-background.png", "ice_cave/darkcave-background.png"},
-  {"darkcave-middle.png", "ice_cave/darkcave-middle.png"},
-  {"darkcave-preview.png", "ice_cave/darkcave-preview.png"},
-  {"darkcave-top_and_bottom.png", "ice_cave/darkcave-top_and_bottom.png"},
-  {"darkcavemidground-middle.png", "ice_cave/darkcavemidground-middle.png"},
-  {"darkcavemidground-top_and_bottom.png", "ice_cave/darkcavemidground-top_and_bottom.png"},
-  {"black_800px.png", "misc/black_800px.png"},
-  {"fog.png", "misc/fog.png"},
-  {"grid.png", "misc/grid.png"},
-  {"grid.surface", "misc/grid.surface"},
-  {"heatshimmer-displacement.png", "misc/heatshimmer-displacement.png"},
-  {"heatshimmer.png", "misc/heatshimmer.png"},
-  {"heatshimmer.surface", "misc/heatshimmer.surface"},
-  {"leaves.png", "misc/leaves.png"},
-  {"oiltux.jpg", "misc/oiltux.jpg"},
-  {"transparent_up.png", "misc/transparent_up.png"},
-  {"nightsky.png", "nightsky/nightsky.png"},
-  {"nightsky_bottom.png", "nightsky/nightsky_bottom.png"},
-  {"nightsky_middle.png", "nightsky/nightsky_middle.png"},
-  {"nightsky_para.png", "nightsky/nightsky_para.png"},
-  {"nightsky_top.png", "nightsky/nightsky_top.png"},
-};
-
-} // namespace
-
-SurfacePtr
-Background::load_background(const std::string& image_path)
+void
+Background::set_top_image_action(const std::string& action)
 {
-  if (image_path.empty())
-    return nullptr;
+  if (m_image_top)
+    m_image_top->set_action(action);
+}
 
-  if (PHYSFS_exists(image_path.c_str()))
-    // No need to search fallback paths.
-    return Surface::from_file(image_path);
+void
+Background::set_image_action(const std::string& action)
+{
+  m_image->set_action(action);
+}
 
-  // Search for a fallback image in fallback_paths.
-  const std::string& default_dir = "images/background/";
-  const std::string& default_dir2 = "/images/background/";
-  std::string new_path = image_path;
-  if (image_path.substr(0, default_dir.length()) == default_dir)
-    new_path.erase(0, default_dir.length());
-  else if (image_path.substr(0, default_dir2.length()) == default_dir2)
-    new_path.erase(0, default_dir2.length());
-  auto it = fallback_paths.find(new_path);
-  if (it == fallback_paths.end())
-    // Unknown image, let the texture manager select the dummy texture.
-    return Surface::from_file(image_path);
+void
+Background::set_bottom_image_action(const std::string& action)
+{
+  if (m_image_bottom)
+    m_image_bottom->set_action(action);
+}
 
-  new_path = default_dir + it->second;
-  return Surface::from_file(new_path);
+void
+Background::set_image_actions(const std::string& action)
```

Comment:
I'm not sure about the name of this method. It's too similar to the other one.

Reply:
Not sure how else to name it better. Also, if someone has to debug something in the code, I believe they'd notice the additional `s` at the end.